.github/workflows/count_changed_files.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+"""Script to limit number of file changes in single PR.
+
+Methodology:
+
+    Analyses the Pull request to find if the count of file changed in a PR
+    exceeds a pre-defined nummber 20
+
+    This scripts encourages contributors to align with project practices,
+    reducing the likelihood of unintentional merges into incorrect branches.
+
+NOTE:
+
+    This script complies with our python3 coding and documentation standards.
+    It complies with:
+
+        1) Pylint
+        2) Pydocstyle
+        3) Pycodestyle
+        4) Flake8
+
+"""
+
+import os
+import sys
+
+
+def _count_changed_files(base_branch, current_branch):
+    """
+    Count the number of changed files between two branches.
+
+    Args:
+        base_branch (str): The base branch.
+        current_branch (str): The current branch.
+
+    Returns:
+        int: The number of changed files.
+
+    Raises:
+        SystemExit: If an error occurs during execution.
+    """
+    try:
+        base_branch = f"origin/{base_branch}"
+        current_branch = f"origin/{current_branch}"
+        # Run git command to get the list of changed files
+        command = f"git diff --name-only {base_branch}...{current_branch} | wc -l"
```

Comment:
1. Use `subprocess.Popen.communicate`. 
1. It will give you better error messaging if the command fails. 
    - https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate